### PR TITLE
complex comments download and API call 

### PR DESCRIPTION
**Pull request by asegun-cod**
add functionality to download complex comments
add working code to call API
refactor data upload functionality

### CODE REVIEW

File: R/mod_data_management.R [hunk 1]
```diff
@@ -114,18 +128,17 @@ mod_data_management_server <- function(id, db_conn, filter_data) {
...
 
     # Edit a row and effect it in the UI view ####
 
     observeEvent(input$pat_table_cell_edit, {
-      
-      info <- input$pat_table_cell_edit         # get the edited row information
+      info <- input$pat_table_cell_edit # get the edited row information
 
-      info$value <- sapply(info$value, html_decoder, USE.NAMES = F) # decode any html character introduced to the values 
+      info$value <- sapply(info$value, html_decoder, USE.NAMES = F) # decode any html character introduced to the values
```
Comment (ChrisBeeley) [inline R/mod_data_management.R:141]: Please use `FALSE` and not `F`

File: R/mod_data_management.R [hunk 2]
```diff
@@ -303,50 +316,135 @@ mod_data_management_server <- function(id, db_conn, filter_data) {
...
+        #   icon = icon('comment')
+        # )
+      }
+    })
+
+    output$complex_com <- downloadHandler(
+      filename = paste0("complex_comments-", Sys.Date(), ".csv"),
+      content = function(file) {
+        withProgress(message = "Downloading...", value = 0, {
+          write.csv(dt_out$complex_comments, file, row.names = FALSE)
```
Comment (ChrisBeeley) [inline R/mod_data_management.R:349]: Is csv the best format? Wouldn't Word be better?

File: R/tidy_upload.R [hunk 3]
```diff
@@ -15,133 +45,125 @@ clean_dataframe <- function(data){
...
+  dplyr::rows_append(
+    dplyr::tbl(conn, trust_id),
+    final_df,
+    copy = TRUE,
+    in_place = TRUE
+  )
+  
+  print('Done appending to database ...')
+  
+  # reset the db data ------------------- remove later ------------------------
```
Comment (ChrisBeeley) [inline R/tidy_upload.R:161]: There is functionality in golem where you have different configuration for dev versus prod, so you could think about using that here

**Review by ChrisBeeley — COMMENTED** [review]
I've added some comments. Nothing urgent so merge it in whenever you like

File: R/mod_data_management.R (hunk 2 above)
Comment (asegun-cod) [reply R/mod_data_management.R:349]: I am giving users the functionality to download a dataframe with all the rows with complex comments alongside the columns (including the predicted labels) for them to interrogate the data. 

Do you suggest we write only the comments to a .doc file?

File: R/tidy_upload.R (hunk 3 above)
Comment (asegun-cod) [reply R/tidy_upload.R:161]: I can't figure this out for now. I will deal with it  later

File: R/mod_data_management.R (hunk 1 above)
Comment (asegun-cod) [reply R/mod_data_management.R]: fix here https://github.com/CDU-data-science-team/experiencesdashboard/pull/134/commits/015d9b6cbacc92928eb8de50050c43f9410664a9

**Pull request merged**